lib/ui/ui_dart_state.cc

```diff
@@ -214,4 +222,18 @@ bool UIDartState::enable_skparagraph() const {
   return enable_skparagraph_;
 }
 
+void UIDartState::HandlePlatformMessage(
+    std::unique_ptr<PlatformMessage> message) {
+  if (platform_configuration_) {
```

Comment:
To be more direct about my comment:

I think @jason-simmons' feedback woul dbe addressed if this was changed to something like

```
  if (platform_message_handler_)
```

Where `platform_message_handler_` is a `std::weak_ptr`. Alternatively, it could be a `fml::WeakPtr` and we'd just have to post a task to the right task runner here.

Reply:
Yea I think that's a good idea.  I suggested something similar to Jason.  I'll look into it.

Reply:
Okay, I got a chance to give it a shot today.  I think it worked out well.  It's more clean since it doesn't change the threading policy for existing functions except the PlatformMessageHandler which will be threadsafe anyways since it won't be present except for embedders that support background platform channels.  Since PlatformMessageHandlers are already using std::shared_ptr I can use std::weak_ptr safely.  I think this is the winning approach.  I'll clean everything up this afternoon and put everything into review.